pyproj/_crs.pyx

```diff
@@ -215,97 +473,439 @@ cdef class Ellipsoid:
+        grid.package_name = decode_or_undefined(out_package_name)
+        grid.url = decode_or_undefined(out_url)
+        return grid
+
+    def __str__(self):
+        return self.full_name
+
+    def __repr__(self):
+        return ("Grid(short_name={short_name}, full_name={full_name}, package_name={package_name}, "
+                "url={url}, direct_download={direct_download}, open_license={open_license}, "
+                "available={available})").format(
+            short_name=self.short_name,
+            full_name=self.full_name,
+            package_name=self.package_name,
+            url=self.unit_name,
+            direct_download=self.direct_download,
+            open_license=self.open_license,
+            available=self.available
+        )
+
+
+cdef class CoordinateOperation(Base):
+    """
+    Coordinate operation for CRS.
+
+    Attributes
+    ----------
+    name: str
+        The name of the method(projection) with authority information.
+    method_name: str
+        The method (projection) name.
+    method_auth_name: str
+        The method authority name.
+    method_code: str
+        The method code.
+    is_instantiable: int
+        If 1, a coordinate operation can be instantiated as a PROJ pipeline.
+        This also checks that referenced grids are available. 
+    has_ballpark_transformation: int
+        If 1, the coordinate operation has a “ballpark” transformation, 
+        that is a very approximate one, due to lack of more accurate transformations. 
+    accuracy: float
+        The accuracy (in metre) of a coordinate operation. 
+
+    """
+    def __cinit__(self):
+        self._params = None
+        self._grids = None
+        self.method_name = "undefined"
+        self.method_auth_name = "undefined"
+        self.method_code = "undefined"
+        self.is_instantiable = False
+        self.has_ballpark_transformation = False
+        self.accuracy = float("nan")
+
+    @staticmethod
+    cdef create(PJ_CONTEXT* projcontext, PJ* projobj):
+        cdef CoordinateOperation coord_operation = CoordinateOperation()
+        coord_operation.projobj = proj_crs_get_coordoperation(
+            projcontext,
+            projobj
+        )
+        if coord_operation.projobj == NULL:
+            return None
+
+        cdef char *out_method_name = NULL
+        cdef char *out_method_auth_name = NULL
+        cdef char *out_method_code = NULL
+        proj_coordoperation_get_method_info(
+            projcontext,
+            coord_operation.projobj,
+            &out_method_name,
+            &out_method_auth_name,
+            &out_method_code
+        )
+        coord_operation.method_name = decode_or_undefined(out_method_name)
+        coord_operation.method_auth_name = decode_or_undefined(out_method_auth_name)
+        coord_operation.method_code = decode_or_undefined(out_method_code)
+        coord_operation.name = "{method_name} [{method_auth_name}:{method_code}]".format(
```

Comment:
A CoordinateOperation can actually also have a name, eg:

```
In [36]: crs = pyproj.CRS.from_epsg(31370)                                                                                                                    

In [37]: crs.coordinate_operation.name                                                                                                                        
Out[37]: 'Lambert Conic Conformal (2SP) [EPSG:9802]'

In [38]: crs.coordinate_operation                                                                                                                             
Out[38]: 
CONVERSION["Belgian Lambert 72",
    METHOD["Lambert Conic Conformal (2SP)",
        ID["EPSG",9802]],
    PARAMETER["Latitude of false origin",90,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8821]],
    PARAMETER["Longitude of false origin",4.36748666666667,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8822]],
    PARAMETER["Latitude of 1st standard parallel",51.1666672333333,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8823]],
    PARAMETER["Latitude of 2nd standard parallel",49.8333339,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8824]],
    PARAMETER["Easting at false origin",150000.013,
        LENGTHUNIT["metre",1],
        ID["EPSG",8826]],
    PARAMETER["Northing at false origin",5400088.438,
        LENGTHUNIT["metre",1],
        ID["EPSG",8827]],
    ID["EPSG",19961]]
```
In the first line of the WKT, there is "Belgian Lambert 72", which is the "name" of this EPSG:19961 coordinate operation. 
So maybe we should call the above something else? Or only set it the "name" is not defined? (I suppose this name of the CoordinateOperation can be get with `proj_get_name`)

Reply:
The name in the coordinate operation is stored in `method_name`. The `name` is carries over from `Base`, so I filled it with a name that seemed useful for `CRS.__repr__`.

Reply:
> The name in the coordinate operation is stored in `method_name`. The `name` is carries over from `Base`, so I filled it with a name that seemed useful for `CRS.__repr__`.

`method_name` is not the name of the coordinate operation. It is the name of the projection method, as far as I understand the WKT. 
In the example above "Belgian Lambert 72" is the 'name' of the CoordinateOperation (conversion), and "Lambert Conic Conformal (2SP)" is the name of the method (the actual projection)

Reply:
Ah, I see what you mean. Definitely something to look into. Thanks for pointing that out!

Reply:
FIxed - I must have mixed this up with CoordinateSystem. Thanks!